### ISSUE TITLE
GATK restricted issues

### ISSUE THREAD

**Issue by brrk** [chapmanb/bcbio-nextgen#1467]
Hi all,

I have the latest stable version of bcbio, to which I've linked, via the toolplus command, version 3.4 (free) of the GATK. When I attempt to run an exome analysis, everything goes through okay until it comes time to call haplotypes, when I obtain this error: 

` "Require full version of GATK 2.4+ for haplotype calling"
`

The culprit line is `variation/gatk.py:100:`
```
 assert broad_runner.gatk_type() == "restricted", \
            "Require full version of GATK 2.4+ for haplotype calling"
```

When I comment this out, things get a little further ahead, but then this error is thrown: 

```
[...]Annotation DepthOfCoverage is no longer available in the GATK [...]
```

Once again, an elementary spelunk at `variation/annotation.py:27` helps push things along. 

So far, after these hacks, HaplotypeCaller is running smoothly. But I'm wondering if there's a reason for these particular lines in the code. After all, HaplotypeCaller is supported by the free version of GATK (I've run plenty of analyses with it outside bcbio), so why the assertion in line 100 of `gatk.py`? 

Disclaimer: I've read through some of the forum material as well as the Installation guide. I understand that there's an issue linking to third-party software that manifests as both open- and closed-source. But I don't see why any issues should exist *after* having run the toolplus command. What am I missing?

thanks,

rk

**Comment by chapmanb** [comment]
Thanks for the report and sorry about the issue. It looks like bcbio thinks you have an older version of GATK so doesn't make the correct adjustments. If you look at `/path/to/bcbio/manifest/toolplus-packages.yaml`, what version of gatk does it think you have? Perhaps there was some problem with toolplus installation or something else during the install?

Practically, the checks in the code are correct. GATK 2.4+ and above have restricted licensing and they can only be used freely by academic users, so HaplotypeCaller is only available in these versions. The second error is similar, in that the newer versions changed the name from `DepthOfCoverage` to `Coverage`. If you can identify why bcbio does not recognize your version that should help resolve both problems without needing to change the code. Hope this helps.

**Comment by brrk** [comment]
Hi! Thanks so much for the prompt reply. I really appreciate the work your team is doing here. It's very  convenient to have so many different---and useful!---tools in one place, and to be able run them with a few straightforward commands. 

To the issue: The path `/path/to/bcbio/manifest/toolplus-packages.yaml` does not exist on my machine. Specifically, when I ran toolplus, a new folder named `upgrade` was created in the directory immediately above `/path/to/bcbio/`. It was in here that the following directory structure was put in place:
 ```
parent

           anaconda

           toolplus 

           config

           manifest

           galaxy
```

Since this directory didn't have  a `bin` subdirectory, I assumed that I needed to copy the newly created `yaml` files in `galaxy` to the `path/to/bcbio/galaxy/` folder and forget altogether about the new `upgrade` folder created during the upgradation process.

Upon further inspection, however, I now find that the `bin` directory I need is inside the `anaconda` folder, itself a subdirectory of the `upgrade` folder. Further, the file you mention, namely, `toolplus-packages.yaml`, exists in the `manifest` subdirectory of the `upgrade` folder. 

Given the above facts, I'm guessing that I need to point my `PATH` to the `bin` directory inside `upgrade/anaconda` and `rm -rf /path/to/bcbio`. 

Is this correct? 

If so, I'm wondering why the upgrade process didn't directly upgrade the `/path/to/bcbio/bin` and instead created a new directory entirely. The command I ran to upgrade bcbio was identical to the one I ran to _install_ bcbio, except this one had the upgrade command, as well as the path to GATK. 

`bcbio_nextgen_install.py upgrade --tooldir=/path/to/bcbio --toolplus gatk=/path/to/GenomeAnalysisTK.jar`

To summarize, there isn't a huge problem anymore, but one that mostly bears on how I may do future upgrades.

**Comment by chapmanb** [comment]
Thanks for the details. The issue is that you should be running the upgrade with the installed `bcbio_nextgen.py`, not the `bcbio_nextgen_install.py` script. The later installs a new version of bcbio in the directory `upgrade`.

So if you do:
```
bcbio_nextgen.py upgrade --toolplus gatk=/path/to/GenomeAnalysisTK.jar
```
it should do the right thing. 

More generally, you shouldn't need to do any manual copying, removing with bcbio as all should be automated. So if you're finding yourself thinking you need to do that something is wrong.

Hope this helps get everything sorted out.

**Comment by brrk** [comment]
Thanks. I knew I was doing something wrong.
